topgun/k8s/container_limits_test.go

```diff
@@ -0,0 +1,99 @@
+package k8s_test
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/onsi/gomega/gbytes"
+	"github.com/onsi/gomega/gexec"
+
+	. "github.com/concourse/concourse/topgun"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func testLimits() {
+	buildSession := fly.Start("execute", "-c", "../tasks/tiny.yml")
+	<-buildSession.Exited
+	Expect(buildSession.ExitCode()).To(Equal(0))
+
+	hijackSession := fly.Start(
+		"hijack",
+		"-b", "1",
+		"--", "sh", "-c",
+		"cat /sys/fs/cgroup/memory/memory.memsw.limit_in_bytes; cat /sys/fs/cgroup/cpu/cpu.shares",
+	)
+	<-hijackSession.Exited
+
+	Expect(hijackSession.ExitCode()).To(Equal(0))
+	Expect(hijackSession).To(gbytes.Say("1073741824\n512"))
+
+}
+
+var _ = Describe("Garden Config", func() {
+	var (
+		proxySession        *gexec.Session
+		releaseName         string
+		namespace           string
+		atcEndpoint         string
+		nodeImage           string
+		helmDeployTestFlags []string
+	)
+
+	BeforeEach(func() {
+		releaseName = fmt.Sprintf("topgun-cl-%d-%d", randomGenerator.Int(), GinkgoParallelNode())
+		namespace = releaseName
+		Run(nil, "kubectl", "create", "namespace", namespace)
+	})
+
+	JustBeforeEach(func() {
+		helmDeployTestFlags = []string{
+			`--set=worker.replicas=1`,
+			`--set=concourse.web.defaultTaskCpuLimit=512`,
+			`--set=concourse.web.defaultTaskMemoryLimit=1GB`,
+			"--set=worker.nodeSelector.nodeImage=" + nodeImage,
+		}
+		deployConcourseChart(releaseName, helmDeployTestFlags...)
+
+		waitAllPodsInNamespaceToBeReady(namespace)
+
+		By("Creating the web proxy")
+		proxySession, atcEndpoint = startPortForwarding(namespace, "service/"+releaseName+"-web", "8080")
+
+		By("Logging in")
+		fly.Login("test", "test", atcEndpoint)
+
+		Eventually(func() []Worker {
+			return getRunningWorkers(fly.GetWorkers())
+		}, 2*time.Minute, 10*time.Second).
+			ShouldNot(HaveLen(0))
+	})
+
+	AfterEach(func() {
+		helmDestroy(releaseName)
+		Wait(Start(nil, "kubectl", "delete", "namespace", namespace, "--wait=false"))
+		Wait(proxySession.Interrupt())
+	})
+
+	Context("passing a config map location to the worker to be used by gdn", func() {
+		BeforeEach(func() {
+			nodeImage = "cos"
+		})
+
+		It("returns the configure default container limit", func() {
+			testLimits()
+		})
+	})
+
+	XContext("passing a config map location to the worker to be used by gdn", func() {
```

Comment:
In the Baggageclaim drivers tests, we were instead asserting the negation (e.g., asserting that it fails to deploy) so that the test continuously run despite the fact that we expect the `false` condition in that case.

With an `XContext` we don't have that behaviour though - we'd need to manually deactivate `XContext` to see if the negative in such case still holds true.

Wdyt of going with an assertion that verifies that it doesn't work on `ubuntu`?

Reply:
makes sense. Deployment doesn't fail in this case, though. Running the task is the one that fails.